logical/framework/field_data.go

```diff
@@ -252,9 +262,18 @@ func (d *FieldData) getPrimitive(k string, schema *FieldSchema) (interface{}, bo
 
 	case TypeSlice:
 		var result []interface{}
-		if err := mapstructure.WeakDecode(raw, &result); err != nil {
-			return nil, true, err
+
+		switch inp := raw.(type) {
+		case string:
```

Comment:
This will certainly break existing TypeSlice implementations.  I currently don't see any implementations but they could exist elsewhere.  Additionally, this would make this slice implementation slight different from the rest of the implementations.  

WeakDecode will take a string and convert it to a slice which is the desired behavior here because the CLI doesn't know what the actual type is.  Now this will be bypassed and a test like this will fail but succeeds in the current implementation:

```
		"slice type, single string": {
			map[string]*FieldSchema{
				"foo": &FieldSchema{Type: TypeSlice},
			},
			map[string]interface{}{
				"foo": "bar",
			},
			"foo",
			[]interface{}{"bar"},
		},
```

This example would be from the command line like this `vault write path/key foo=bar` on a TypeSlice field.

Reply:
Yeah. Changing the current types is listed as an abandoned option in the RFC. I will add your suggestion of TypeJSONSlice and TypeJSONMap. I think that's a reasonable option.